engine/src/io/data_parser/sql/AbstractSQLParser.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2021 BlazingDB, Inc.
+ *     Copyright 2021 Percy Camilo Triveño Aucahuasi <percy@blazingdb.com>
+ */
+
+#ifndef _ABSTRACTSQLPARSER_H_
+#define _ABSTRACTSQLPARSER_H_
+
+#include "io/data_parser/DataParser.h"
+#include "arrow/io/interfaces.h"
+#include <memory>
+#include <vector>
+
+namespace ral {
+namespace io {
+
+class abstractsql_parser : public data_parser {
```

Comment:
What is the purpose of this class? It looks almost the same as the original data_parser interface?

Reply:
for that matter, it looks like its not used anywhere. Please delete it if its not being used

Reply:
`abstractsql_parser` is a _thing_ now, it allow us to define some common flows for all the sql parsers

Reply:
done